src/ruby/lib/grpc/generic/active_call.rb

```diff
@@ -325,15 +325,16 @@ def each_remote_read_then_finish
       begin
         loop do
           resp = remote_read
-          if resp.nil?  # the last response was received
-            receive_and_check_status
-            break
-          end
+          break if resp.nil?  # the last response was received
```

Comment:
this looks mostly good to me, but I have a cosmetic nit:

instead of wrapping the entire read loop in a `rescue` clause, can we instead wrap only `remote_read` in, for example, a separate function that itself has the `rescue CallError` clause? Such a function could return nil in the case that a `CallError` was raised, or when `remote_read` itself returned nil. This would be consistent with how this sort of thing is handled in the bidi call [read stream enumerable](https://github.com/grpc/grpc/blob/master/src/ruby/lib/grpc/generic/bidi_call.rb#L198).

thanks!

Reply:
I think this implementation would be more efficient than your suggestion.
If the code wraps only `remote_read`, it calls unnecessary  `begin ~ rescue ~ end` every `remote_read`.
An also becuase it has not had consistency so far, I think this code would be ok as it is. What do you think?

>  I am a little busy and am slow to review the other PR's you have sent right now, but will get to them soon.

Please do not worry about it :smile:. but, Would you give me the write permission to this repository?
I have a plan to create other PR's. If you have little time to review, it'll be very helpful to you.

Reply:
> I think this implementation would be more efficient than your suggestion.
If the code wraps only remote_read, it calls unnecessary begin ~ rescue ~ end every remote_read.

I'm not really worried about the efficiency of this method right now. And I'd much prefer to use small-scoped `begin` and `rescue` blocks rather for simplicity. If there are people complaining about the performance of this call, <i>and</i> there is a benchmark that shows that the `begin`/`rescue` block are causing a large penalty, then I'd be open to optimizations, but that's not the scenario currently :)

> Please do not worry about it 😄. but, Would you give me the write permission to this repository?

Our repo has a policy that requires every PR to get approved, even if you have write access.

Reply:
@ganmacs this change would be good to have, but please change to reduce the scope of the `begin`/`rescue` block

Reply:
OK, I changed the scope of the being ~ rescue block.